src/main/java/org/moltimate/moltimatebackend/util/ProteinUtils.java

```diff
@@ -35,7 +41,25 @@ public static Structure queryPdb(String pdbId) {
                 return MMCIF_FILE_READER.getStructureById(pdbId);
             } catch (IOException mmcifReaderError) {
                 mmcifReaderError.printStackTrace();
-                throw new RuntimeException("Cannot find structure with PDB id " + pdbId);
+                throw new InvalidPdbIdException(pdbId);
+            }
+        }
+    }
+
+    /**
+     * Use this when processing a list so that an error doesn't interrupt execution
+     * @param pdbId PDB ID string
+     * @return Optional of a BioJava Structure object representing the PDB ID
+     */
+    public static Optional<Structure> queryPdbOptional(String pdbId) {
+        try {
+            return Optional.of(PDB_FILE_READER.getStructureById(pdbId));
+        } catch (IOException pdbReaderError) {
+            try {
+                return Optional.of(MMCIF_FILE_READER.getStructureById(pdbId));
+            } catch (IOException mmcifReaderError) {
+                log.error("Cannot find structure with PDB id " + pdbId);
```

Comment:
Removing and changing to print one log.error with a list of entries that failed (which would occur as an alternative to throwing the error to avoid redundancy)